pkg/tests/upgrade_tests/shapshot.go

```diff
@@ -0,0 +1,259 @@
+	}
+	for _, schema := range upgradedDbInfo.schemas {
+		expected, ok := pristineSchemas[schema.name]
+		if !ok {
+			t.Logf("extra schema %s", schema.name)
+			continue
+		}
+		tablesDiff := schema.tables != expected.tables
+		functionsDiff := schema.functions != expected.functions
+		privilegesDiff := schema.privileges != expected.privileges
+		indicesDiff := schema.indices != expected.indices
+		triggersDiff := schema.triggers != expected.triggers
+		dataDiff := !reflect.DeepEqual(schema.data, expected.data)
+		if tablesDiff || functionsDiff || privilegesDiff || indicesDiff || triggersDiff || dataDiff {
+			t.Logf("differences in schema: %s", schema.name)
+		}
+		if tablesDiff {
+			t.Logf("tables\nexpected:\n\t%s\ngot:\n\t%s", expected.tables, schema.tables)
+		}
+		if functionsDiff {
+			t.Logf("functions\nexpected:\n\t%s\ngot:\n\t%s", expected.functions, schema.functions)
+		}
+		if privilegesDiff {
+			t.Logf("privileges\nexpected:\n\t%s\ngot:\n\t%s", expected.privileges, schema.privileges)
+		}
+		if indicesDiff {
+			t.Logf("indices\nexpected:\n\t%s\ngot:\n\t%s", expected.indices, schema.indices)
+		}
+		if triggersDiff {
+			t.Logf("triggers\nexpected:\n\t%s\ngot:\n\t%s", expected.triggers, schema.triggers)
+		}
+		if dataDiff {
+			t.Logf("data\nexpected:\n\t%+v\ngot:\n\t%+v", expected.data, schema.data)
+		}
+	}
+}
+
+var replaceChildren = regexp.MustCompile("timescaledb_internal\\._hyper_.*\n")
+
+func GetDbInfo(t *testing.T, container testcontainers.Container, dbName, outputDir string, db *pgxpool.Pool, useTimescaleDB bool) (info dbInfo) {
+	info.schemaNames = getSchemas(t, db)
+	considerSchemas := schemas
+	if !useTimescaleDB {
+		considerSchemas = schemasWOTimescaleDB
+	}
+	if !reflect.DeepEqual(info.schemaNames, considerSchemas) {
+		t.Errorf(
+			"unexpected schemas.\nexpected\n\t%v\ngot\n\t%v",
+			considerSchemas,
+			info.schemaNames,
+		)
+	}
+
+	info.extensions = getPsqlInfo(t, container, dbName, outputDir, "\\dx")
+	info.schemas = make([]schemaInfo, len(ourSchemas))
+	for i, schema := range ourSchemas {
+		info := &info.schemas[i]
+		info.name = schema
+		info.tables = getPsqlInfo(t, container, dbName, outputDir, "\\d+ "+schema+".*")
+		info.tables = replaceChildren.ReplaceAllLiteralString(info.tables, "timescaledb_internal._hyper_*\n")
+		info.functions = getPsqlInfo(t, container, dbName, outputDir, "\\df+ "+schema+".*")
+		info.privileges = getPsqlInfo(t, container, dbName, outputDir, "\\dp "+schema+".*")
+		// not using \di+ since the sizes are too noisy, and the descriptions
+		// will be in tables anyway
+		info.indices = getPsqlInfo(t, container, dbName, outputDir, "\\di "+schema+".*")
+		info.triggers = getPsqlInfo(t, container, dbName, outputDir, "\\dy "+schema+".*")
+		info.data = getTableInfosForSchema(t, db, schema)
+	}
+	return
+}
+
+func GetDbInfoIgnoringTable(t *testing.T, container testcontainers.Container, dbName, outputDir string, db *pgxpool.Pool, ignoreTable string, useTimescaleDB bool) (info dbInfo) {
```

Comment:
I'd prefer the table to be schema qualified
```suggestion
func GetDbInfoIgnoringTable(t *testing.T, container testcontainers.Container, dbName, outputDir string, db *pgxpool.Pool, ignoreTableSchema string, ignoreTableName string, useTimescaleDB bool) (info dbInfo) {
```